### PR TITLE
Sync insiders to discord and discourse

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Build algorithm
- Add insiders_status field to users table
- Symbolize insiders_status
- Add command to update insiders_status
- Add commands for granting and revoking access
- Add joined insiders email
- Add activate insiders command
- Recheck active insiders job
- Check insiders status when subscription changes
- Trigger when roles change
- Trigger when reputation changes
- Trigger when reputation changes
- Add unset
- Rename
- Refactor
- Rename command
- Update from commands
- Remove after_commit trigger
- Move command
- Add missing file
- Add notifications for lifetime insiders and expired insiders
- Various tweaks and improvemnets
- Support feature flag for insiders (#3666)
- Setup project
- Add header and features content and raw spacement
- Finish basic styling
- We're aiming font-medium
- Add pixel decorations
- Update announcement-bar, site-header
- Update signed off header
- Adjust site header icons and filters
- Update site footer
- Add color instead of none
- Remove hidden box-shadow
- Add insiders status prompt (#3671)
- CSS Tweaks
- Tweak copy
- Fix tests
- Pipe things together
- Various more tweaks
- Add locks
- Fix locks
- Add notifications for joining
- Enable tests
- Fix issue with unset overwriting status
- Tweak logic
- Send notification when becoming eligible
- Update more notifications
- Only set status to unset if currently ineligible
- Recheck eligible and active insiders
- Set insider discord roles
- Add insiders group in discourse
- Create separate method for adding user to group
- Create constants for discourse group names
- Add insiders group in discourse
- Fix tests

### CODE REVIEW

File: app/commands/user/set_discord_roles.rb [hunk 1]
```diff
@@ -23,6 +24,12 @@ def set_supermentor_role!
     set_role!(SUPERMENTOR_ROLE_ID)
   end
 
+  def set_insiders_role!
+    return unless user.insider?
+
+    set_role!(INSIDERS_ROLE_ID)
+  end
```
Comment (iHiD) [inline app/commands/user/set_discord_roles.rb:37]: We need the inverse of this too. If they're not an insider any more, we need to remove the role.

File: app/commands/user/set_discourse_groups.rb [hunk 2]
```diff
@@ -21,16 +22,33 @@ def set_trust_level!
...
+    client.group_add(group_id(group_name), user_id: [discourse_user_id])
+  rescue DiscourseApi::UnprocessableEntity
+    # If the user was already a member of the group,
+    # ignore the error
+  end
+
+  def remove_from_group!(group_name)
+    client.group_remove(group_id(group_name), user_id: [discourse_user_id])
+  rescue DiscourseApi::UnprocessableEntity
+    # If the user was already a member of the group,
```
Comment (iHiD) [inline app/commands/user/set_discourse_groups.rb]:
```suggestion
    # If the user was not a member of the group,
```

File: app/commands/user/insiders_status/activate.rb [hunk 3]
```diff
@@ -19,5 +19,7 @@ def call
 
     User::Notification::Create.(user, @notification_key) if FeatureFlag::INSIDERS
     AwardBadgeJob.perform_later(user, :insider) if FeatureFlag::INSIDERS
+    User::SetDiscordRoles.(user)
+    User::SetDiscourseGroups.(user)
```
Comment (ErikSchierboom) [inline app/commands/user/insiders_status/activate.rb:23]: This is done outside of the locking on `user`

File: app/commands/user/insiders_status/update.rb [hunk 4]
```diff
@@ -24,6 +24,8 @@ def call
       end
     end
 
+    User::SetDiscordRoles.(user)
+    User::SetDiscourseGroups.(user)
```
Comment (ErikSchierboom) [inline app/commands/user/insiders_status/update.rb:28]: This is done outside the `user` lock

**Pull request merged**